frontends/web/src/routes/device/bitbox02/bitbox02.tsx

```diff
@@ -373,18 +359,12 @@ class BitBox02 extends Component<Props, State> {
       title: this.props.t('bitbox02Interact.followInstructionsMnemonicTitle'),
       text: this.props.t('bitbox02Interact.followInstructionsMnemonic'),
     } });
-    apiPost('devices/bitbox02/' + this.props.deviceID + '/restore-from-mnemonic').then(({ success }) => {
-      if (!success) {
+    restoreFromMnemonic(this.props.deviceID)
+      .then(() => this.setState({ appStatus: 'restoreFromMnemonic' }))
+      .catch(() => {
         alertUser(this.props.t('bitbox02Wizard.restoreFromMnemonic.failed'), { asDialog: false });
-      } else {
-        this.setState({
-          appStatus: 'restoreFromMnemonic',
-        });
-      }
-      this.setState({
-        waitDialog: undefined,
-      });
-    });
+      })
+      .finally(() => this.setState({ waitDialog: undefined }));
```

Comment:
finally is called after the promise is settled, either resoved or rejected

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally